src/go/rpk/pkg/cli/topic/describe_storage.go

```diff
@@ -0,0 +1,258 @@
+			}
+			// By default, if neither are specified, we opt in to
+			// the size section only
+			if !summary && !size && !sync && !count && !offset {
+				summary, size = true, true
+			}
+			if all {
+				summary, size, sync, count, offset = true, true, true, true, true
+			}
+			var sections int
+			for _, b := range []bool{summary, size, sync, count, offset} {
+				if b {
+					sections++
+				}
+			}
+			withSections := sections > 1
+
+			header("SUMMARY", summary, withSections, func() {
+				tw := out.NewTabWriter()
+				defer tw.Flush()
+				tw.PrintColumn("NAME", topic)
+				if topicMeta.IsInternal {
+					tw.PrintColumn("INTERNAL", topicMeta.IsInternal)
+				}
+				tw.PrintColumn("PARTITIONS", len(topicMeta.Partitions))
+				if len(topicMeta.Partitions) > 0 {
+					p0 := &topicMeta.Partitions[0]
+					tw.PrintColumn("REPLICAS", len(p0.Replicas))
+				}
+				// We can safely pick the value of the first partition since
+				// the cloud storage mode doesn't change between partitions of
+				// the same topic.
+				tw.PrintColumn("CLOUD-STORAGE-MODE", report[0].CloudStatus.CloudStorageMode)
+
+				lastUpload := getLastUpload(report)
+				tw.PrintColumn("LAST-UPLOAD", humanReadable(lastUpload, human, "time"))
+			})
+
+			header("LOG-SIZE", size, withSections, func() {
+				tw := out.NewTable("PARTITION", "TOTAL", "CLOUD", "LOCAL")
+				defer tw.Flush()
+				for _, r := range report {
+					tw.Print(
+						r.Partition,
+						humanReadable(r.CloudStatus.TotalLogSize, human, "size"),
+						humanReadable(r.CloudStatus.CloudLogSize, human, "size"),
+						humanReadable(r.CloudStatus.LocalLogSize, human, "size"),
+					)
+				}
+			})
+
+			header("SYNC", sync, withSections, func() {
+				isRrr := report[0].CloudStatus.CloudStorageMode == "read_replica"
+				headers := []string{"PARTITION", "LAST-SEGMENT-UPLOAD", "LAST-MANIFEST-UPLOAD", "METADATA-UPDATE-PENDING"}
+				if isRrr {
+					headers = append(headers, "LAST-MANIFEST-SYNC")
+				}
+				tw := out.NewTable(headers...)
+				defer tw.Flush()
+				for _, r := range report {
+					row := []any{
+						r.Partition,
+						humanReadable(r.CloudStatus.LastSegmentUpload, human, "time"),
+						humanReadable(r.CloudStatus.LastManifestUpload, human, "time"),
+						r.CloudStatus.MetadataUpdatePending,
+					}
+					// If isRrr is true, LastManifestSync should be there, this
+					// nil check is possibly redundant, but we want to avoid a
+					// nil pointer dereference.
+					if isRrr && r.CloudStatus.LastManifestSync != nil {
+						row = append(row, humanReadable(*r.CloudStatus.LastManifestSync, human, "time"))
+					}
+					tw.Print(row...)
+				}
+			})
+
+			header("LOG-SEGMENT-COUNT", count, withSections, func() {
+				tw := out.NewTable("PARTITION", "CLOUD", "LOCAL")
+				defer tw.Flush()
+				for _, r := range report {
+					tw.Print(r.Partition, r.CloudStatus.CloudLogSegmentCount, r.CloudStatus.LocalLogSegmentCount)
+				}
+			})
```

Comment:
wdyt about merging this in with the LOG-SIZE section, so that the columns become:
```
PARTITION  CLOUD-SEGMENTS  CLOUD-BYTES  LOCAL-SEGMENTS  LOCAL-BYTES
```
which has some symmetry with the LOG-OFFSET section?

Reply:
(or going with the alternative approach of merging by local & cloud)

Reply:
Merged LOG-SEGMENT-COUNT with SIZE section